controllers/vspherecluster_controller.go

```diff
@@ -194,6 +194,16 @@ func (r clusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr er
 func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconcile.Result, error) {
 	ctx.Logger.Info("Reconciling VSphereCluster delete")
 
+	vsphereMachines, err := infrautilv1.GetVSphereMachinesInCluster(ctx, ctx.Client, ctx.VSphereCluster.Namespace, ctx.VSphereCluster.Name)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err,
+			"unable to list machines part of VSphereCluster %s/%s", ctx.VSphereCluster.Namespace, ctx.VSphereCluster.Name)
+	}
+
+	if len(vsphereMachines) > 0 {
```

Comment:
note: this should be enough, I don't think users are expecting us to Garbage collect `VSphereMachines` If they delete the `VSphereCluster`

Reply:
If I delete a `Cluster`, that will clean up everything.

If I keep my `Cluster` around, and just delete the `VSphereCluster`, it'll stick around until I manually delete my `VSphereMachines` - is that correct? If so, we should also consider an event here.

Reply:
Yes, correct. Although I don't expect users to do it directly, the way this case could happen is to delete the whole namespace. I plan to raise a ticket for Eventing in CAPV, so will address this in a follow-up